### PR TITLE
Mitaka nfp 13 may 2016 vpn agent fixes

### PR DESCRIPTION
**Pull request by sdharav**
@njagadish  please review the changes.

### CODE REVIEW

File: gbpservice/nfp/service_plugins/vpn/drivers/nfp_vpnaas_driver.py [hunk 1]
```diff
@@ -256,3 +253,6 @@ def create_vpnservice(self, context, vpnservice):
             rsrc_id=vpnservice['id'],
             resource=vpnservice,
             reason='create', service_vendor=service_vendor)
+
+    def delete_vpnservice(self, context, vpnservice):
```
Comment (njagadish) [inline gbpservice/nfp/service_plugins/vpn/drivers/nfp_vpnaas_driver.py:257]: Why is this added? Is this sent from service driver?

File: gbpservice/nfp/service_plugins/vpn/drivers/nfp_vpnaas_driver.py [hunk 2]
```diff
@@ -188,8 +183,8 @@ def create_ipsec_site_connection(self, context, ipsec_site_connection):
                                     context,
                                     ipsec_site_connection['vpnservice_id'])
 
-        starttime = endtime = time.time()
-        while(endtime - starttime) < TIMEOUT:
```
Comment (njagadish) [inline gbpservice/nfp/service_plugins/vpn/drivers/nfp_vpnaas_driver.py:192]: Any reason why the old logic is modified. That should also work right?

File: gbpservice/nfp/service_plugins/vpn/drivers/nfp_vpnaas_driver.py [hunk 3]
```diff
@@ -69,11 +69,6 @@ def update_status(self, context, status):
         plugin = self.driver.service_plugin
         plugin.update_status_by_agent(context, status)
 
-    def ipsec_site_conn_deleted(self, context, ipsec_site_conn_id):
```
Comment (njagadish) [inline gbpservice/nfp/service_plugins/vpn/drivers/nfp_vpnaas_driver.py:72]: Will the DB get cleaned without this callback? Why is this removed?

File: gbpservice/nfp/service_plugins/vpn/drivers/nfp_vpnaas_driver.py (hunk 1 above)
Comment (sdharav) [inline gbpservice/nfp/service_plugins/vpn/drivers/nfp_vpnaas_driver.py:257]: actually we get call delete-vpn-service from heat engine, because of which we were getting that extra call with empty kwargs which breaking in config_orch's vpn agent so here we are suppressing that.

File: gbpservice/nfp/service_plugins/vpn/drivers/nfp_vpnaas_driver.py (hunk 2 above)
Comment (sdharav) [inline gbpservice/nfp/service_plugins/vpn/drivers/nfp_vpnaas_driver.py:192]: just to get rid of one extra variable

File: gbpservice/nfp/service_plugins/vpn/drivers/nfp_vpnaas_driver.py (hunk 3 above)
Comment (sdharav) [inline gbpservice/nfp/service_plugins/vpn/drivers/nfp_vpnaas_driver.py:72]: yes Jagadish,  it cleans before making the call to configurator.
